packaging/resource_suite_s3_nocache.py

```diff
@@ -2869,3 +2867,149 @@ def test_iput_large_file_with_checksum_issue_2124(self):
             self.user0.assert_icommand("irm -f {file1}".format(**locals()), 'EMPTY')
             s3plugin_lib.remove_if_exists(file1)
             s3plugin_lib.remove_if_exists(file1_get)
+
+class Test_S3_NoCache_Decoupled_Base(Test_S3_NoCache_Base):
+
+    def test_decoupled_redirect_issue_2146(self):
```

Comment:
How is this triggering a redirect? Are there multiple iRODS servers in the zone?

Reply:
I'm not sure whether plugin tests can be made topology-aware (perhaps when running manually), but the `resource_host` is set to `test.settings.HOSTNAME_3` which would guarantee that the client would not be connected to that host in the tests. But like you say, that only applies if `test.settings.ICAT_HOSTNAME`, `test.settings.HOSTNAME_1`, and `test.settings.HOSTNAME_3` are referring to different servers.

And there is of course no harm in testing it on a single machine as well. :)

Reply:
As Alan said, this could be run in topology it would trigger a redirect.  When running on one host it will not trigger the redirect.

I added it just in case that functionality was available in the future.